src/libs/fee/getL1Fee.ts

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { BigNumber, ethers } from "ethers";
+import { EstimateL1FeeParams } from "../types";
+import { Multicall, ContractCallContext, ContractCallResults } from "ethereum-multicall";
+
+const ABI = {
+  Optimism: [
+    "function getL1GasUsed(bytes executeData) returns (uint256)",
+    "function scalar() returns (uint256)",
+    "function overhead() returns (uint256)",
+  ],
+  Mantle: ["function overhead() returns (uint256)", "function scalar() returns (uint256)"],
+};
+
+/**
+ * Get the estimated L1 fee for a given L2 chain.
+ * @param env The environment to use. Either "mainnet" or "testnet".
+ * @param chain The destination L2 chain.
+ * @param params The parameters to use for the estimation.
+ * @returns The estimated L1 fee.
+ */
+export function getL1FeeForL2(
+  provider: ethers.providers.JsonRpcProvider,
+  chain: string,
+  params: EstimateL1FeeParams
+): Promise<BigNumber> {
+  const multicall = new Multicall({ ethersProvider: provider, tryAggregate: true });
+
+  switch (chain) {
```

Comment:
is there a generalized way of doing this? for example, we are adding fraxtal (L2 op-stack), blast (L2 op-stack), and I can see this quickly leading to a bug if we forget to bump the sdk version for new chains we add and don't update this switch statement

Reply:
yeah, i think we can dynamically handle L2 fee calculation by exposing L2 flags (e.g. `L2_type`: `op | arb | mantle | undefined`). Then, the sdk can calculate the L2 fee based on those flags.

Reply:
@npty , updated in this PR: https://github.com/axelarnetwork/gmp-api/pull/571, please review.

Reply:
@nrsirapop awesome, thanks you! 🙏🏻

Reply:
@canhtrinh incorporated `l2_type` from `getFees` api in [this commit](https://github.com/axelarnetwork/axelarjs-sdk/pull/299/commits/956540f408311b6a83ddc635468f7348ab763a68)